terraform/ecs/main.tf

```diff
@@ -58,21 +58,120 @@ provider "aws" {
 data "aws_caller_identity" "current" {
 }
 
-module "ecs_cluster" {
-  source  = "infrablocks/ecs-cluster/aws"
-  version = "4.0.0"
-
-  cluster_name                         = module.common.testing_id
-  component                            = "aoc"
-  deployment_identifier                = "testing"
-  vpc_id                               = module.basic_components.aoc_vpc_id
-  subnet_ids                           = module.basic_components.aoc_private_subnet_ids
-  region                               = var.region
-  associate_public_ip_addresses        = "yes"
-  security_groups                      = [module.basic_components.aoc_security_group_id]
-  cluster_desired_capacity             = 1
-  cluster_instance_iam_policy_contents = file("instance-policy.json")
-  // TODO(pingleig): pass patch tag for canary and soaking (if any)
+resource "aws_ecs_cluster" "ecscluster" {
+  name = module.common.testing_id
+
+  depends_on = [
+    null_resource.iam_wait
+  ]
+}
+
+resource "aws_ecs_cluster_capacity_providers" "clustercapacity" {
+  cluster_name = aws_ecs_cluster.ecscluster.name
+
+  capacity_providers = [aws_ecs_capacity_provider.capacityprovider.name]
+
+}
+
+resource "aws_ecs_capacity_provider" "capacityprovider" {
+  name = "capacityprovider"
+
+  auto_scaling_group_provider {
+    auto_scaling_group_arn = aws_autoscaling_group.clusterasg.arn
+  }
+}
+
+data "aws_ami" "amazon_linux_2" {
+  most_recent = true
+  owners      = ["amazon"]
+
+  filter {
+    name   = "name"
+    values = ["amzn2-ami-ecs-hvm-*-x86_64-ebs"]
+  }
+}
+
+resource "aws_launch_template" "launchtemp" {
+  name_prefix   = "launchtemp"
```

Comment:
do we need to add tags so that ssm knows how to patch this?
https://github.com/aws-observability/aws-otel-test-framework/blob/terraform/terraform/ec2/main.tf#L95

if yes, we also need to wait for the patching to finish before moving on with the tests.

Reply:
I've added these tags to the launch template instances in my latest commit

Reply:
FYI that's not how the SSM patching is triggered in ec2. The `check_patch` resource uses that variable as a flag. https://github.com/aws-observability/aws-otel-test-framework/blob/80f205ee27e1b60c7e110fdc14ddda8385bc1808/terraform/ec2/main.tf#L143